DownFall/Views/HUD.swift

```diff
@@ -52,6 +52,10 @@ class HUD: SKSpriteNode {
                 showAttack(attackInput: input, endTiles: trans.endTiles)
             case .collectItem(_, let item, let total):
                 incrementCurrencyCounter(item, total: total)
+            case .itemUsed:
+                if let tiles = trans.endTiles, let playerCoord = getTilePosition(.player(.zero), tiles: tiles), case TileType.player(let data) = tiles[playerCoord].type {
```

Comment:
This is such a common pattern.  Get the optional tiles, get the optinol player position and unwrap that data.  Can we move it into an extension?

Reply:
not sure how to do that